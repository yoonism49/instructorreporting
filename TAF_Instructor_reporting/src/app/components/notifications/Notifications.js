// @flow weak

import React              from 'react';
import NotificationPanel  from './notificationPanel/NotificationPanel';
import Notification       from './notification/Notification';

const Notifications = () => (
  <NotificationPanel>
    <Notification type={'danger'}>
      <span>
        <strong>
          Oh snap!
        </strong>
        Change a few things up and try submitting again.
      </span>
    </Notification>

    <Notification type={'success'}>
      <span>
        <strong>
          Well done!
        </strong>
        You successfully read this important alert message.
      </span>
    </Notification>

    <Notification type={'info'}>
      <span>
        <strong>
          Heads up!
        </strong>
        This alert needs your attention, but it's not super important.
      </span>
    </Notification>

    <Notification type={'warning'}>
      <span>
        <strong>
          Warning!
        </strong>
        Best check yo self, you're not looking too good.
      </span>
    </Notification>

    <Notification type={'danger'}>
      <span>
        <strong>
          Oh snap!
        </strong>
        Change a few things up and try submitting again.
      </span>
    </Notification>

  </NotificationPanel>
);

export default Notifications;
