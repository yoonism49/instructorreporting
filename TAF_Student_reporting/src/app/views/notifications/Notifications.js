// @flow weak

import React, {
  PureComponent
}                         from 'react';
import PropTypes          from 'prop-types';
import {
  AnimatedView,
  Panel,
  NotificationPanel,
  Notification
}                         from '../../components';
import Highlight          from 'react-highlight';


class Notifications extends PureComponent {
  static propTypes= {
    actions: PropTypes.shape({
      enterNotifications: PropTypes.func.isRequired,
      leaveNotifications: PropTypes.func.isRequired
    })
  };

  componentWillMount() {
    const { actions: { enterNotifications } } = this.props;
    enterNotifications();
  }

  componentWillUnmount() {
    const { actions: { leaveNotifications } } = this.props;
    leaveNotifications();
  }

  render() {
    const source = `
      // import
      import { NotificationPanel, Notification } from './_SOMEWHERE_/components';

      // in render():
      <div className="row">
        <div className="col-md-4 col-md-offset-4">
          <NotificationPanel title="notifications">
              <Notification type={'danger'}>
                <span>
                  <strong>
                    Oh snap!
                  </strong>
                  Change a few things up and try submitting again.
                </span>
              </Notification>

              <Notification type={'success'}>
                <span>
                  <strong>
                    Well done!
                  </strong>
                  You successfully read this important alert message.
                </span>
              </Notification>

              <Notification type={'info'}>
                <span>
                  <strong>
                    Heads up!
                  </strong>
                  This alert needs your attention, but it's not super important.
                </span>
              </Notification>

              <Notification type={'warning'}>
                <span>
                  <strong>
                    Warning!
                  </strong>
                  Best check yo self, you're not looking too good.
                </span>
              </Notification>

              <Notification type={'danger'}>
                <span>
                  <strong>
                    Oh snap!
                  </strong>
                  Change a few things up and try submitting again.
                </span>
              </Notification>

          </NotificationPanel>
        </div>
      </div>
      `;

    return(
      <AnimatedView>
        {/* preview: */}
        <div className="row">
          <div className="col-xs-12">
            <Panel
              title="Notifications"
              hasTitle={true}
              bodyBackGndColor={'#F4F5F6'}>
              <div className="row">
                <div className="col-md-4 col-md-offset-4">

                  <NotificationPanel title="notifications">
                      <Notification type={'danger'}>
                        <span>
                          <strong>
                            Oh snap!
                          </strong>
                          Change a few things up and try submitting again.
                        </span>
                      </Notification>

                      <Notification type={'success'}>
                        <span>
                          <strong>
                            Well done!
                          </strong>
                          You successfully read this important alert message.
                        </span>
                      </Notification>

                      <Notification type={'info'}>
                        <span>
                          <strong>
                            Heads up!
                          </strong>
                          This alert needs your attention, but it's not super important.
                        </span>
                      </Notification>

                      <Notification type={'warning'}>
                        <span>
                          <strong>
                            Warning!
                          </strong>
                          Best check yo self, you're not looking too good.
                        </span>
                      </Notification>

                      <Notification type={'danger'}>
                        <span>
                          <strong>
                            Oh snap!
                          </strong>
                          Change a few things up and try submitting again.
                        </span>
                      </Notification>

                  </NotificationPanel>

                </div>
              </div>
            </Panel>
          </div>
        </div>
        {/* source: */}
        <div className="row">
          <div className="col-xs-12">
            <Panel
              title="Source"
              hasTitle={true}>
              <Highlight className="javascript">
                {source}
              </Highlight>
            </Panel>
          </div>
        </div>
      </AnimatedView>
    );
  }
}

export default Notifications;
