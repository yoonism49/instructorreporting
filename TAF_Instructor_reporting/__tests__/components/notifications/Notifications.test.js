
'use strict';

import React              from 'react';
import Notifications      from '../../../src/app/components/notifications/Notifications';
import renderer       from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme

describe('Notifications component', () => {
  it('renders as expected', () => {
    const component = renderer.create(
      <div>
        <Notifications />
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
