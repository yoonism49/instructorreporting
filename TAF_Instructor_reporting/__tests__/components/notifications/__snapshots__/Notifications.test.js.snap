// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications component renders as expected 1`] = `
<div>
  <section
    className="panel"
  >
    <header
      className="panel-heading"
    >
      Notifications
    </header>
    <div
      className="panel-body notification--body-container"
      id="noti-box"
    >
      <div
        className="alert alert-danger"
      >
        <button
          className="close close-sm"
          data-dismiss="alert"
          type="button"
        >
          <i
            className="fa fa-times"
          />
        </button>
        <span>
          <strong>
            Oh snap!
          </strong>
          Change a few things up and try submitting again.
        </span>
      </div>
      <div
        className="alert alert-success"
      >
        <button
          className="close close-sm"
          data-dismiss="alert"
          type="button"
        >
          <i
            className="fa fa-times"
          />
        </button>
        <span>
          <strong>
            Well done!
          </strong>
          You successfully read this important alert message.
        </span>
      </div>
      <div
        className="alert alert-info"
      >
        <button
          className="close close-sm"
          data-dismiss="alert"
          type="button"
        >
          <i
            className="fa fa-times"
          />
        </button>
        <span>
          <strong>
            Heads up!
          </strong>
          This alert needs your attention, but it's not super important.
        </span>
      </div>
      <div
        className="alert alert-warning"
      >
        <button
          className="close close-sm"
          data-dismiss="alert"
          type="button"
        >
          <i
            className="fa fa-times"
          />
        </button>
        <span>
          <strong>
            Warning!
          </strong>
          Best check yo self, you're not looking too good.
        </span>
      </div>
      <div
        className="alert alert-danger"
      >
        <button
          className="close close-sm"
          data-dismiss="alert"
          type="button"
        >
          <i
            className="fa fa-times"
          />
        </button>
        <span>
          <strong>
            Oh snap!
          </strong>
          Change a few things up and try submitting again.
        </span>
      </div>
    </div>
  </section>
</div>
`;
